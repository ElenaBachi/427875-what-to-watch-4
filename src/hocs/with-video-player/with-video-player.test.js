import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withVideoPlayer from "./with-video-player.jsx";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

it(`withVideoPlayer should render correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped>
      <video />
    </MockComponentWrapped>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
