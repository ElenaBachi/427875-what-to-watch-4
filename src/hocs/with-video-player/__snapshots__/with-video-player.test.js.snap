// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideoPlayer should render correctly 1`] = `
<div>
  <video />
</div>
`;
